src/node.rs

```diff
@@ -929,10 +935,12 @@ impl<K, V> TreeBin<K, V> {
     /// method.
     pub(crate) unsafe fn defer_drop_without_values<'g>(
         bin: Shared<'g, BinEntry<K, V>>,
-        guard: &'g Guard,
+        guard: &'g Guard<'_>,
     ) {
-        guard.defer_unchecked(move || {
-            if let BinEntry::Tree(mut tree_bin) = *bin.into_owned().into_box() {
+        guard.retire(bin.as_ptr(), |mut link| {
+            let bin = link.cast::<BinEntry<K, V>>();
+
+            if let BinEntry::Tree(mut tree_bin) = Linked::into_inner(*Box::from_raw(bin)) {
```

Comment:
Am I blind? Where's the `unsafe {}` for `Box::from_raw`?

Reply:
Also, can't this use `bin.into_box()`?

Reply:
```rust
unsafe {
    let _ = || {
        std::mem::uninitialized::<()>();
    };
}
```
compiles.

Reply:
I'll change it to `into_box` :+1:

Reply:
That's _wild_ that that compiles. I hate that so much. At the very least we should enable https://github.com/rust-lang/rust/issues/71668, though I know that doesn't fully fix this instance.

Reply:
We can't actually use `into_box`, because this is a `*mut Linked<T>`, not a `Shared<'_, T>`, but I did add the safety comments.

Reply:
Enabling \`unsafe_op_in_unsafe_fn\` causes 34 new errors, do you think we can do it in a separate PR :sweat_smile:

Reply:
Hehe, yep, that's fine :+1: